src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -3639,34 +3697,91 @@ public ServiceConfig getConfig() {
   }
 
   /**
-   * loads a yaml configuration file from the file system default location will
-   * be data/config/{configName}/{name}.yml
+   * load a single service entry into the plan through yml or default
    * 
    * @param name
+   * @param type
    * @return
    * @throws IOException
    */
-  static public ServiceConfig load(String name) throws IOException {
+  private ServiceConfig loadService(Plan inPlan, String name, String type, Boolean overwrite, Boolean overwritePeers) throws IOException {
+    // FIXME - get default if file doesn't exist !
+    // FIXME - special handling for runtime
+    // ServiceInterface si = create(name);
+    if (overwrite == null) {
+      overwrite = false;
+    }
 
-    ServiceInterface si = create(name);
-    Runtime runtime = Runtime.getInstance();
+    if (overwritePeers == null) {
+      overwritePeers = false;
+    }
+
+    if (inPlan == null) {
+      // root plan
+      plan = new Plan(runtime.serviceType);
+    } else {
+      plan = inPlan;
+    }
+
+    if (name == null && type == null) {
+      log.info("no new name or type information provided - we will use existing plan");
+      log.info("looking for runtime config");
+      ServiceConfig sc = plan.get("runtime");
+      if (sc != null) {
+        log.info("found runtime config");
+        return sc;
+      } else {
+        log.warn("did not find runtime config");
+        return null;
+      }
+    }
+
+    ServiceConfig originalSc = plan.get(name);
+    if (!overwrite && originalSc != null) {
+      log.info("will not overwrite and plan entry already exists for {}", name);
+      return originalSc;
+    }
+
+    // if config name set and yaml file exists - it takes precedence
     String filename = runtime.getConfigDir() + fs + runtime.getConfigName() + fs + name + ".yml";
+    File check = new File(filename);
+    ServiceConfig sc = null;
+    if (runtime.getConfigName() != null && check.exists()) {
+      sc = CodecUtils.readServiceConfig(filename);
+    }
 
-    ServiceConfig config = CodecUtils.readServiceConfig(filename);
-    si.load(config);
+    if (sc != null) {
+      log.info("{} found yml file - loading into plan", name);
+      runtime.plan.put(name, sc);
+      return sc;
+    } else {
+      log.info("no {}.yml found", name);
+    }
 
-    if (runtime != null) {
-      runtime.invoke("publishConfigLoaded", name);
+    if (sc == null) {
+      // no file based yml config
+      if (type == null) {
+        error("%s could not load type %s", name, type);
+        return null;
+      } else {
+        log.info("could not find config in file trying default of type {}", type);

```

Comment:
should this be a warning/error?  the log message tends to indicate some problem.. maybe re-word the message so it doesn't sound like an error

Reply:
its an error - couldn't get type information for the requested service